torchvision/models/shufflenetv2.py

```diff
@@ -0,0 +1,186 @@
+            pw_conv11(inp if (self.stride > 1) else branch_features, branch_features),
+            nn.BatchNorm2d(branch_features),
+            nn.ReLU(inplace=True),
+            dw_conv33(branch_features, branch_features),
+            nn.BatchNorm2d(branch_features),
+            pw_conv11(branch_features, branch_features),
+            nn.BatchNorm2d(branch_features),
+            nn.ReLU(inplace=True),
+        )
+
+    @staticmethod
+    def depthwise_conv(i, o, kernel_size, stride=1, padding=0, bias=False):
+        return nn.Conv2d(i, o, kernel_size, stride, padding, bias=bias, groups=i)
+
+    def forward(self, x):
+        if self.stride == 1:
+            x1, x2 = x.chunk(2, dim=1)
+            out = torch.cat((x1, self.branch2(x2)), dim=1)
+        else:
+            out = torch.cat((self.branch1(x), self.branch2(x)), dim=1)
+
+        out = channel_shuffle(out, 2)
+
+        return out
+
+
+class ShuffleNetV2(nn.Module):
+    def __init__(self, num_classes=1000, input_size=224, width_mult=1):
+        super(ShuffleNetV2, self).__init__()
+
+        try:
+            self.stage_out_channels = self._getStages(float(width_mult))
+        except KeyError:
+            raise ValueError('width_mult {} is not supported'.format(width_mult))
+
+        input_channels = 3
+        output_channels = self.stage_out_channels[0]
+        self.conv1 = nn.Sequential(
+            nn.Conv2d(input_channels, output_channels, 3, 2, 1, bias=False),
+            nn.BatchNorm2d(output_channels),
+            nn.ReLU(inplace=True),
+        )
+        input_channels = output_channels
+
+        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+
+        stage_names = ['stage{}'.format(i) for i in [2, 3, 4]]
+        stage_repeats = [4, 8, 4]
+        for name, repeats, output_channels in zip(
+                stage_names, stage_repeats, self.stage_out_channels[1:]):
+            seq = [InvertedResidual(input_channels, output_channels, 2)]
+            for i in range(repeats - 1):
+                seq.append(InvertedResidual(output_channels, output_channels, 1))
+            setattr(self, name, nn.Sequential(*seq))
+            input_channels = output_channels
+
+        output_channels = self.stage_out_channels[-1]
+        self.conv5 = nn.Sequential(
+            nn.Conv2d(input_channels, output_channels, 1, 1, 0, bias=False),
+            nn.BatchNorm2d(output_channels),
+            nn.ReLU(inplace=True),
+        )
+
+        if (input_size % 32):
+            raise ValueError('illegal input_size')
+        self.globalpool = nn.AvgPool2d(int(input_size / 32))
+
+        # expected ifm size is: channels x 1 x 1
+        self.fc = nn.Linear(self.stage_out_channels[-1], num_classes)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.maxpool(x)
+        x = self.stage2(x)
+        x = self.stage3(x)
+        x = self.stage4(x)
+        x = self.conv5(x)
+        x = self.globalpool(x)
```

Comment:
I think we should replace `globalpool` + `view` with a `x.mean([2, 3])`.

Reply:
thanks for the advice!